client/src/components/river/index.tsx

```diff
@@ -35,17 +38,44 @@ const qSourcesAndResources = loader(
 );
 
 const FhirRiverView = (): React.ReactElement => {
+  const dispatch = useDispatch();
   const toaster = useSelector((state: IReduxStore) => state.toaster);
+  const { data: batchList, error: batchListError } = useSelector(getBatchList);
 
   const [selectedSource, setSelectedSource] = useState({} as Source);
   const [selectedResources, setSelectedResources] = useState([] as Resource[]);
   const [running, setRunning] = useState(false);
+  const [selectedBatch, setSelectedBatch] = useState('');
 
   const { data } = useQuery(qSourcesAndResources, {
     fetchPolicy: 'no-cache'
   });
 
+  useEffect(() => {
+    dispatch(listBatch());
```

Comment:
This will only happen once when the component is mounted, right? Is it what we want?
Do we need something to refresh the list of batches?

Reply:
Agreed!
I'm just worried that batches that have terminated will appear but it's not that severe